File: app/src/main/java/com/arjanvlek/oxygenupdater/Model/UpdateMethod.java
package com.arjanvlek.oxygenupdater.Model;

import com.fasterxml.jackson.annotation.JsonProperty;

public class UpdateMethod {

    private long id;
    private String englishName;
    private String dutchName;
    private boolean recommended;

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    @JsonProperty("english_name")
    public String getEnglishName() {
        return englishName;
    }

    @JsonProperty("english_name")
    public void setEnglishName(String englishName) {
        this.englishName = englishName;
    }

    @JsonProperty("dutch_name")
    public String getDutchName() {
        return dutchName;
    }

    @JsonProperty("dutch_name")
    public void setDutchName(String dutchName) {
        this.dutchName = dutchName;
    }

    public boolean isRecommended() {
        return recommended;
    }

    public void setRecommended(String recommended) {
        this.recommended = (recommended != null && recommended.equals("1"));
    }
}

File: app/src/main/java/com/arjanvlek/oxygenupdater/models/UpdateMethod.kt
package com.arjanvlek.oxygenupdater.models

import com.fasterxml.jackson.annotation.JsonProperty

data class UpdateMethod(
    override val id: Long,
    val englishName: String?,
    val dutchName: String?,
    var recommended: Boolean = false,
    val recommendedForRootedDevice: Boolean = false,
    val recommendedForNonRootedDevice: Boolean = false,
    val supportsRootedDevice: Boolean = false
) : SelectableModel {

    override val name = englishName

    @JsonProperty("recommended")
    fun setRecommended(recommended: String?): UpdateMethod {
        this.recommended = recommended != null && recommended == "1"
        return this
    }
}
